tw-tkms-starter/src/main/java/com/transferwise/kafka/tkms/TransactionalKafkaMessageSender.java

```diff
@@ -396,13 +395,23 @@ protected void validateMessage(TkmsMessage message, int messageIdx) {
           message.getShard(), properties.getShardsCount());
     }
     Preconditions.checkNotNull(message.getValue(), "%s: Value can not be null.", messageIdx);
+    boolean uuidHeaderPresent = false;
     if (message.getHeaders() != null) {
       for (int headerIdx = 0; headerIdx < message.getHeaders().size(); headerIdx++) {
         Header header = message.getHeaders().get(headerIdx);
         Preconditions.checkNotNull(header.getValue(), "%s: Header value @{%s} can not be null.", messageIdx, headerIdx);
         Preconditions.checkArgument(!Strings.isNullOrEmpty(header.getKey()), "%s: Header key @{%s} can not be null.", messageIdx, headerIdx);
+        uuidHeaderPresent |= StandardHeaders.X_WISE_UUID.equals(header.getKey());
       }
     }
+    if (properties.isUuidHeaderRequired() && !uuidHeaderPresent) {
+      throw new IllegalArgumentException(
```

Comment:
I think we need to have a bit more info in the error message how to fix this or point it specifically to changelog or some doc. This would be critical if that somebody gets this upgrade while dealing with an incident.

I don't think we currently have a good practice of going over changelogs.

Reply:
Dropped the requirement. Can be implemented separately from this change